src/librustc_typeck/check/mod.rs

```diff
@@ -819,14 +819,10 @@ pub fn check_item_body<'a,'tcx>(ccx: &CrateCtxt<'a,'tcx>, it: &'tcx hir::Item) {
                     check_const(ccx, trait_item.span, &expr, trait_item.id)
                 }
                 hir::MethodTraitItem(ref sig, Some(ref body)) => {
```

Comment:
But that will cause ICEs in erroneous code - the error in the validation pass is not fatal and compilation continues until type checking is done.